lib/tower_cli/models/base.py

```diff
@@ -914,5 +919,99 @@ def monitor(self, pk, min_interval=1, max_interval=30,
+        """Print the current job status. This is used to check a running job.
+        You can look up the job with the same parameters used for a get
+        request."""
+        # Remove default values (anything where the value is None).
+        self._pop_none(kwargs)
+
+        # Search for the record if pk not given
+        if not pk:
+            job = self.get(include_debug_header=True, **kwargs)
+        # Get the job from Ansible Tower if pk given
+        else:
+            debug.log('Asking for job status.', header='details')
+            finished_endpoint = '%s%d/' % (self.endpoint, pk)
+            job = client.get(finished_endpoint).json()
+
+        # In most cases, we probably only want to know the status of the job
+        # and the amount of time elapsed. However, if we were asked for
+        # verbose information, provide it.
+        if detail:
+            return job
+
+        # Print just the information we need.
+        return adict({
+            'elapsed': job['elapsed'],
+            'failed': job['failed'],
+            'status': job['status'],
+        })
+
+    @resources.command
+    @click.option('--fail-if-not-running', is_flag=True, default=False,
+                  help='Fail loudly if the job is not currently running.')
+    def cancel(self, pk=None, fail_if_not_running=False, **kwargs):
+        """Cancel a currently running job.
+
+        Fails with a non-zero exit status if the job cannot be canceled.
+        You must provide either a pk or parameters in the job's identity.
+        """
+        # Search for the record if pk not given
+        if not pk:
+            existing_data = self.get(**kwargs)
+            pk = existing_data['id']
+
+        cancel_endpoint = '%s%d/cancel/' % (self.endpoint, pk)
+        # Attempt to cancel the job.
+        try:
+            client.post(cancel_endpoint)
+            changed = True
+        except exc.MethodNotAllowed:
+            changed = False
+            if fail_if_not_running:
+                raise exc.TowerCLIError('Job not running.')
+
+        # Return a success.
+        return adict({'status': 'canceled', 'changed': changed})
+
+
+class Resource(ResourceMethods):
+    """This is the parent class for all 'standard' resources."""
+    abstract = True
+
+    @resources.command
+    @click.option('--fail-on-found', default=False,
+                  show_default=True, type=bool, is_flag=True,
+                  help='If used, return an error if a matching record already '
+                       'exists.')
+    @click.option('--force-on-exists', default=False,
+                  show_default=True, type=bool, is_flag=True,
+                  help='If used, if a match is found on unique fields, other '
+                       'fields will be updated to the provided values. If '
+                       'False, a match causes the request to be a no-op.')
+    def create(self, fail_on_found=False, force_on_exists=False, **kwargs):
```

Comment:
Found it - this is the part that caused the coverage decline. doc strings on commands are more than just simple suggestions. They are used in the help text on the command line, so the descriptive text should have been moved here. The parsing function was intended to operate on a specific type of string in this help text (which apparently doesn't exist in other, similar, methods), and it doesn't exist because of inadvertent over-riding.